3 create enemy ia

Zulrage opened the pull request:
- Adding monster attack and vision of the player
(No line of sigth, no jumps)

Commit:
Detection zone

- Wrap the world in a PlayWorld object to track players positions
- Set the world in the setWorld method instead of consturcor for
MovableObject
- Add observers
- Remove a box from the menu

Commit:
Add basic IA for player tracking to enemy

Commit:
Add monster attack

- Add a gourdin to monster and the terrrrible ability to attack

Commit:
Fix monster bugs
- Destroy joints before destroying the body
- Adding blocking time to ranged weapons

Xuecnahc commented on core/src/game/shadowlight/entities/Monster.java:
```diff
@@ -1,10 +1,126 @@
+public class Monster extends MovableObject implements IObserver, IMovable {
+
+  // Visibility properties
+  protected boolean faceRight;
```
Direction is life

Xuecnahc replied:
But life has no direction

Xuecnahc commented on core/src/game/shadowlight/entities/Monster.java:
```diff
@@ -1,10 +1,126 @@
+  // Visibility properties
+  protected boolean faceRight;
+  protected boolean alertState = false;
+  protected float visibilityRangeFront = 5, visibilityRangeBack = 0;
```
Please declare each variable on its own line

Xuecnahc commented on core/src/game/shadowlight/screens/PlayScreen.java:
```diff
@@ -81,14 +86,27 @@ else if (player.getBody().getPosition().x > bottomRight.x)
+      movable.move();
+    }
+    debugRenderer.render(playWorld.getWorld(), camera.combined);
+  //  debugRenderer.setDrawJoints(false);
```
Remove comented code

Commit:
Format code
- Use direction
- Remove useless comment
- Return to the line for each variable declaration

Xuecnahc commented:
👍
